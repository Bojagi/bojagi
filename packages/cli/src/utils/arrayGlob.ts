import glob from './glob';

export default async function arrayGlob(
  patterns: string[],
  { executionPath }: { executionPath: string }
) {
  return (
    await Promise.all(patterns.map(storyPath => glob(storyPath, { cwd: executionPath })))
  ).flat();
}
