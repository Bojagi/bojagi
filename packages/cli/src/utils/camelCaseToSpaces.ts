export function camelCaseToSpaces(input: string) {
  return input
    .replace(/([a-z])([A-Z0-9])/g, '$1 $2')
    .replace(/([0-9])([A-Za-z])/g, '$1 $2')
    .split(' ')
    .map(word => word.substr(0, 1).toUpperCase() + word.substr(1))
    .join(' ');
}
