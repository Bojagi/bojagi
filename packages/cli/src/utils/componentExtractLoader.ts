import * as path from 'path';

export function pitch(remainingRequest) {
  // Change the request from an /abolute/path.js to a relative ./path.js
  // This prevents [chunkhash] values from changing when running webpack
  // builds in different directories.
  const newRequestPath = remainingRequest.replace(
    this.resourcePath,
    `.${path.sep}${path.relative(this.context, this.resourcePath)}`
  );
  if (this.cacheable) this.cacheable();
  if (!this.query) throw new Error('query parameter is missing');

  // eslint-disable-next-line no-underscore-dangle
  const moduleName = JSON.stringify(this._module.issuer.name);

  const componentHubFn = 'registerComponent';
  const requirePath = JSON.stringify(`-!${newRequestPath}`);
  const result = `
    var componentModule = require(${requirePath});
    ${componentHubFn}(${moduleName}, componentModule);
    module.exports = componentModule;
  `;

  return result;
}
