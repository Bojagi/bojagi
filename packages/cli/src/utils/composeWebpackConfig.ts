import * as webpack from 'webpack';
import * as pathUtils from 'path';
import { mergeWithCustomize, unique } from 'webpack-merge';

const BLACKLISTED_PLUGINS = ['ManifestPlugin'];

const composeWebpackConfig = (
  baseConfig: webpack.Configuration,
  entry: webpack.Entry,
  executionPath: string,
  decoratorFile: string | undefined,
  getSbOption: <T>(key: string, fallbackValue: T) => T,
  publicPath: string = '__bojagi_public_path__/'
): webpack.Configuration => {
  const { entry: baseEntry, plugins, ...baseConfigWithoutEntry } = baseConfig;
  const filteredPlugins = plugins?.filter(
    plugin => !BLACKLISTED_PLUGINS.includes(plugin.constructor?.name)
  );

  return mergeWithCustomize({
    customizeArray: unique(
      'plugins',
      ['DefinePlugin'],
      plugin => plugin.constructor && plugin.constructor.name
    ),
  })(
    // Bojagi decorator
    {
      module: {
        rules: decoratorFile
          ? [
              {
                test: pathUtils.resolve(executionPath, decoratorFile),
                use: [
                  {
                    loader: `bojagi-expose-loader`,
                    options: {
                      symbol: 'bojagiDecorator',
                    },
                  },
                ],
              },
            ]
          : [],
      },
    },
    {
      ...baseConfigWithoutEntry,
      plugins: filteredPlugins,
    },
    {
      entry,
      output: {
        path: pathUtils.join(process.cwd(), 'bojagi'),
        filename: '[name].js',
        publicPath,
        globalObject: 'window',
      },
      resolve: {
        alias: {
          'storybook-folder': pathUtils.resolve(getSbOption('configDir', './.storybook') || ''),
        },
      },
      resolveLoader: {
        alias: {
          'component-extract-loader': pathUtils.resolve(__dirname, 'componentExtractLoader'),
          'bojagi-expose-loader': pathUtils.resolve(__dirname, 'exposeLoader'),
          reactDom: 'react-dom',
        },
      },
      externals: {
        react: 'react',
        'react-dom': 'reactDom',
      },
      optimization: {
        minimize: true,
        concatenateModules: false,
        splitChunks: {
          cacheGroups: {
            vendors: {
              test: /[\\/]node_modules[\\/]/,
              name: 'vendors',
              chunks: 'all',
            },
          },
        },
      },
      plugins: [
        new webpack.DefinePlugin({
          'process.env': {
            NODE_ENV: JSON.stringify('production'),
          },
        }),
        new webpack.NormalModuleReplacementPlugin(
          /@storybook\/addons/,
          require.resolve('@bojagi/cli/fakeStorybookAddons.js')
        ),
        new webpack.NormalModuleReplacementPlugin(
          /@storybook\/react/,
          require.resolve('@bojagi/cli/fakeStorybookReact.js')
        ),
      ],
    }
  );
};

export default composeWebpackConfig;
