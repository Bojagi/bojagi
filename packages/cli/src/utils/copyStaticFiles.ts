import * as path from 'path';
import { getFS } from '../dependencies';

const fs = getFS();

async function copyStaticFilesFromFolder(folder: string, outputPath: string, subPath: string = '') {
  const folderFiles = await fs.promises.readdir(path.join(folder, subPath));
  return Promise.all(
    folderFiles.map(async folderFile => {
      const relativePath = path.join(subPath, folderFile);
      const fullPath = path.join(folder, relativePath);
      const fullOutputPath = path.join(outputPath, relativePath);
      const stats = await fs.promises.stat(fullPath);

      if (stats.isDirectory()) {
        await fs.promises.mkdir(fullOutputPath);
        return copyStaticFilesFromFolder(folder, outputPath, relativePath);
      }
      await fs.promises.copyFile(fullPath, fullOutputPath);
      return [relativePath];
    })
  ).then(files => files.flat());
}

export async function copyStaticFiles(folders: string[], outputPath: string): Promise<string[]> {
  return Promise.all(
    folders.filter(fs.existsSync).map(folder => copyStaticFilesFromFolder(folder, outputPath))
  ).then(files => files.flat());
}
