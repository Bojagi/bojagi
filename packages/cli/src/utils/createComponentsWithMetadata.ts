import { EntrypointWithMetadata } from '../commands/bundle';
import { Module } from '../types';

export class EntrypointError extends Error {}
export class FilePathError extends Error {}

export default function createComponentsWithMetadata(
  entrypointsWithMetadata: Record<string, EntrypointWithMetadata>,
  compilerOutput: Record<string, string>,
  modules: Module[]
) {
  return Object.entries(entrypointsWithMetadata).reduce((prev, [key, ep]) => {
    const fileContent = compilerOutput[key];
    if (!/^.+?\?.+?\!.+?$/.test(ep.entrypoint)) {
      throw new EntrypointError('Entrypoint does not follow right syntax');
    }

    const fullFilePath = ep.entrypoint.split('!')[1];
    const prefixPath = `${process.cwd()}/`;
    if (!fullFilePath.startsWith(prefixPath)) {
      throw new FilePathError('File does not start with correct path');
    }
    const filePath = fullFilePath.replace(new RegExp(`^${prefixPath}`), '');
    const matchingModule = modules.find(module => module.resource === filePath);
    const dependencies = matchingModule && matchingModule.dependencies;
    const components = ep.components.map(({ symbol, isDefaultExport }) => ({
      fileContent,
      symbol,
      isDefaultExport,
      filePath,
      dependencies,
      exportName: isDefaultExport ? 'default' : symbol,
      name: symbol
    }));
    return [...prev, ...components];
  }, []);
}
