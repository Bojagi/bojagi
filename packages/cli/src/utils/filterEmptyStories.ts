export function filterEmptyStories(story) {
  return !!story.storyItems && story.storyItems.length;
}
