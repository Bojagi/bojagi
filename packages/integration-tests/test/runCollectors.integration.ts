import { deleteTmpFolder, execBojagi, snapShotTmpFolder } from './utils';

test('run main collector', () => {
  deleteTmpFolder();
  execBojagi('bundle');
  execBojagi('runCollectors');
  snapShotTmpFolder();
});
