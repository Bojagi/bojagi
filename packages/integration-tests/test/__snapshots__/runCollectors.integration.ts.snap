// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run main collector 1`] = `
"[
 {
  \\"name\\": \\"commons\\"
 }
]"
`;

exports[`run main collector 2`] = `
"[
 {
  \\"symbol\\": \\"Box\\",
  \\"isDefaultExport\\": true,
  \\"filePath\\": \\"src/components/Box.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Box.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"isExternal\\": true
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"filePath\\": \\"../../node_modules/styled-components/dist/styled-components.browser.esm.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"dependencies\\": [
     {
      \\"request\\": \\"stylis/stylis.min\\",
      \\"filePath\\": \\"../../node_modules/stylis/stylis.min.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"stylis-rule-sheet\\",
      \\"filePath\\": \\"../../node_modules/stylis-rule-sheet/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"react\\",
      \\"isExternal\\": true
     },
     {
      \\"request\\": \\"@emotion/unitless\\",
      \\"filePath\\": \\"../../node_modules/@emotion/unitless/dist/unitless.browser.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"react-is\\",
      \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"./cjs/react-is.production.min.js\\",
        \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/cjs/react-is.production.min.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     },
     {
      \\"request\\": \\"memoize-one\\",
      \\"filePath\\": \\"../../node_modules/memoize-one/dist/memoize-one.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"prop-types\\",
      \\"filePath\\": \\"../../node_modules/prop-types/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"./factoryWithThrowingShims\\",
        \\"filePath\\": \\"../../node_modules/prop-types/factoryWithThrowingShims.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"./lib/ReactPropTypesSecret\\",
          \\"filePath\\": \\"../../node_modules/prop-types/lib/ReactPropTypesSecret.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       }
      ]
     },
     {
      \\"request\\": \\"@emotion/is-prop-valid\\",
      \\"filePath\\": \\"../../node_modules/@emotion/is-prop-valid/dist/is-prop-valid.browser.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"@emotion/memoize\\",
        \\"filePath\\": \\"../../node_modules/@emotion/memoize/dist/memoize.browser.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     },
     {
      \\"request\\": \\"merge-anything\\",
      \\"filePath\\": \\"../../node_modules/merge-anything/dist/index.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"is-what\\",
        \\"filePath\\": \\"../../node_modules/is-what/dist/index.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     }
    ]
   }
  ],
  \\"exportName\\": \\"default\\",
  \\"name\\": \\"Box\\"
 },
 {
  \\"symbol\\": \\"BoxWithButtons\\",
  \\"isDefaultExport\\": true,
  \\"filePath\\": \\"src/components/BoxWithButtons.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/BoxWithButtons.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"isExternal\\": true
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"filePath\\": \\"../../node_modules/styled-components/dist/styled-components.browser.esm.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"dependencies\\": [
     {
      \\"request\\": \\"stylis/stylis.min\\",
      \\"filePath\\": \\"../../node_modules/stylis/stylis.min.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"stylis-rule-sheet\\",
      \\"filePath\\": \\"../../node_modules/stylis-rule-sheet/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"react\\",
      \\"isExternal\\": true
     },
     {
      \\"request\\": \\"@emotion/unitless\\",
      \\"filePath\\": \\"../../node_modules/@emotion/unitless/dist/unitless.browser.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"react-is\\",
      \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"./cjs/react-is.production.min.js\\",
        \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/cjs/react-is.production.min.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     },
     {
      \\"request\\": \\"memoize-one\\",
      \\"filePath\\": \\"../../node_modules/memoize-one/dist/memoize-one.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"prop-types\\",
      \\"filePath\\": \\"../../node_modules/prop-types/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"./factoryWithThrowingShims\\",
        \\"filePath\\": \\"../../node_modules/prop-types/factoryWithThrowingShims.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"./lib/ReactPropTypesSecret\\",
          \\"filePath\\": \\"../../node_modules/prop-types/lib/ReactPropTypesSecret.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       }
      ]
     },
     {
      \\"request\\": \\"@emotion/is-prop-valid\\",
      \\"filePath\\": \\"../../node_modules/@emotion/is-prop-valid/dist/is-prop-valid.browser.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"@emotion/memoize\\",
        \\"filePath\\": \\"../../node_modules/@emotion/memoize/dist/memoize.browser.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     },
     {
      \\"request\\": \\"merge-anything\\",
      \\"filePath\\": \\"../../node_modules/merge-anything/dist/index.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"is-what\\",
        \\"filePath\\": \\"../../node_modules/is-what/dist/index.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     }
    ]
   },
   {
    \\"request\\": \\"./Box\\",
    \\"filePath\\": \\"src/components/Box.js\\",
    \\"gitPath\\": \\"packages/integration-tests/src/components/Box.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"dependencies\\": [
     {
      \\"request\\": \\"react\\",
      \\"isExternal\\": true
     },
     {
      \\"request\\": \\"styled-components\\",
      \\"filePath\\": \\"../../node_modules/styled-components/dist/styled-components.browser.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"stylis/stylis.min\\",
        \\"filePath\\": \\"../../node_modules/stylis/stylis.min.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       },
       {
        \\"request\\": \\"stylis-rule-sheet\\",
        \\"filePath\\": \\"../../node_modules/stylis-rule-sheet/index.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       },
       {
        \\"request\\": \\"react\\",
        \\"isExternal\\": true
       },
       {
        \\"request\\": \\"@emotion/unitless\\",
        \\"filePath\\": \\"../../node_modules/@emotion/unitless/dist/unitless.browser.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       },
       {
        \\"request\\": \\"react-is\\",
        \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/index.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"./cjs/react-is.production.min.js\\",
          \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/cjs/react-is.production.min.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       },
       {
        \\"request\\": \\"memoize-one\\",
        \\"filePath\\": \\"../../node_modules/memoize-one/dist/memoize-one.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       },
       {
        \\"request\\": \\"prop-types\\",
        \\"filePath\\": \\"../../node_modules/prop-types/index.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"./factoryWithThrowingShims\\",
          \\"filePath\\": \\"../../node_modules/prop-types/factoryWithThrowingShims.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": [
           {
            \\"request\\": \\"./lib/ReactPropTypesSecret\\",
            \\"filePath\\": \\"../../node_modules/prop-types/lib/ReactPropTypesSecret.js\\",
            \\"isExternal\\": false,
            \\"isNodeModule\\": false,
            \\"dependencies\\": []
           }
          ]
         }
        ]
       },
       {
        \\"request\\": \\"@emotion/is-prop-valid\\",
        \\"filePath\\": \\"../../node_modules/@emotion/is-prop-valid/dist/is-prop-valid.browser.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"@emotion/memoize\\",
          \\"filePath\\": \\"../../node_modules/@emotion/memoize/dist/memoize.browser.esm.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       },
       {
        \\"request\\": \\"merge-anything\\",
        \\"filePath\\": \\"../../node_modules/merge-anything/dist/index.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"is-what\\",
          \\"filePath\\": \\"../../node_modules/is-what/dist/index.esm.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       }
      ]
     }
    ]
   },
   {
    \\"request\\": \\"./Button\\",
    \\"filePath\\": \\"src/components/Button.js\\",
    \\"gitPath\\": \\"packages/integration-tests/src/components/Button.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"dependencies\\": [
     {
      \\"request\\": \\"react\\",
      \\"isExternal\\": true
     },
     {
      \\"request\\": \\"styled-components\\",
      \\"filePath\\": \\"../../node_modules/styled-components/dist/styled-components.browser.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"stylis/stylis.min\\",
        \\"filePath\\": \\"../../node_modules/stylis/stylis.min.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       },
       {
        \\"request\\": \\"stylis-rule-sheet\\",
        \\"filePath\\": \\"../../node_modules/stylis-rule-sheet/index.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       },
       {
        \\"request\\": \\"react\\",
        \\"isExternal\\": true
       },
       {
        \\"request\\": \\"@emotion/unitless\\",
        \\"filePath\\": \\"../../node_modules/@emotion/unitless/dist/unitless.browser.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       },
       {
        \\"request\\": \\"react-is\\",
        \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/index.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"./cjs/react-is.production.min.js\\",
          \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/cjs/react-is.production.min.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       },
       {
        \\"request\\": \\"memoize-one\\",
        \\"filePath\\": \\"../../node_modules/memoize-one/dist/memoize-one.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       },
       {
        \\"request\\": \\"prop-types\\",
        \\"filePath\\": \\"../../node_modules/prop-types/index.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"./factoryWithThrowingShims\\",
          \\"filePath\\": \\"../../node_modules/prop-types/factoryWithThrowingShims.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": [
           {
            \\"request\\": \\"./lib/ReactPropTypesSecret\\",
            \\"filePath\\": \\"../../node_modules/prop-types/lib/ReactPropTypesSecret.js\\",
            \\"isExternal\\": false,
            \\"isNodeModule\\": false,
            \\"dependencies\\": []
           }
          ]
         }
        ]
       },
       {
        \\"request\\": \\"@emotion/is-prop-valid\\",
        \\"filePath\\": \\"../../node_modules/@emotion/is-prop-valid/dist/is-prop-valid.browser.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"@emotion/memoize\\",
          \\"filePath\\": \\"../../node_modules/@emotion/memoize/dist/memoize.browser.esm.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       },
       {
        \\"request\\": \\"merge-anything\\",
        \\"filePath\\": \\"../../node_modules/merge-anything/dist/index.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"is-what\\",
          \\"filePath\\": \\"../../node_modules/is-what/dist/index.esm.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       }
      ]
     }
    ]
   }
  ],
  \\"exportName\\": \\"default\\",
  \\"name\\": \\"BoxWithButtons\\"
 },
 {
  \\"symbol\\": \\"Button\\",
  \\"isDefaultExport\\": true,
  \\"filePath\\": \\"src/components/Button.js\\",
  \\"gitPath\\": \\"packages/integration-tests/src/components/Button.js\\",
  \\"dependencies\\": [
   {
    \\"request\\": \\"react\\",
    \\"isExternal\\": true
   },
   {
    \\"request\\": \\"styled-components\\",
    \\"filePath\\": \\"../../node_modules/styled-components/dist/styled-components.browser.esm.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"dependencies\\": [
     {
      \\"request\\": \\"stylis/stylis.min\\",
      \\"filePath\\": \\"../../node_modules/stylis/stylis.min.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"stylis-rule-sheet\\",
      \\"filePath\\": \\"../../node_modules/stylis-rule-sheet/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"react\\",
      \\"isExternal\\": true
     },
     {
      \\"request\\": \\"@emotion/unitless\\",
      \\"filePath\\": \\"../../node_modules/@emotion/unitless/dist/unitless.browser.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"react-is\\",
      \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"./cjs/react-is.production.min.js\\",
        \\"filePath\\": \\"../../node_modules/styled-components/node_modules/react-is/cjs/react-is.production.min.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     },
     {
      \\"request\\": \\"memoize-one\\",
      \\"filePath\\": \\"../../node_modules/memoize-one/dist/memoize-one.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": []
     },
     {
      \\"request\\": \\"prop-types\\",
      \\"filePath\\": \\"../../node_modules/prop-types/index.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"./factoryWithThrowingShims\\",
        \\"filePath\\": \\"../../node_modules/prop-types/factoryWithThrowingShims.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": [
         {
          \\"request\\": \\"./lib/ReactPropTypesSecret\\",
          \\"filePath\\": \\"../../node_modules/prop-types/lib/ReactPropTypesSecret.js\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"dependencies\\": []
         }
        ]
       }
      ]
     },
     {
      \\"request\\": \\"@emotion/is-prop-valid\\",
      \\"filePath\\": \\"../../node_modules/@emotion/is-prop-valid/dist/is-prop-valid.browser.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"@emotion/memoize\\",
        \\"filePath\\": \\"../../node_modules/@emotion/memoize/dist/memoize.browser.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     },
     {
      \\"request\\": \\"merge-anything\\",
      \\"filePath\\": \\"../../node_modules/merge-anything/dist/index.esm.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"dependencies\\": [
       {
        \\"request\\": \\"is-what\\",
        \\"filePath\\": \\"../../node_modules/is-what/dist/index.esm.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"dependencies\\": []
       }
      ]
     }
    ]
   }
  ],
  \\"exportName\\": \\"default\\",
  \\"name\\": \\"Button\\"
 }
]"
`;

exports[`run main collector 3`] = `"c48edc2d38bf80980f4a319da3af6776543458f767d5f1888015b5e35d738430"`;

exports[`run main collector 4`] = `"f0b000b21d8a3d7dbd5e6c7707a326ec6eee0acbfec3d010c6d1b8cdbebd7609"`;

exports[`run main collector 5`] = `
"[
 {
  \\"propSet\\": {
   \\"color\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"yellow\\"
   },
   \\"children\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"I am a yellow Box!\\"
   }
  },
  \\"createdBy\\": \\"Bojagi Collector\\",
  \\"name\\": \\"Yellow\\"
 },
 {
  \\"propSet\\": {
   \\"color\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"green\\"
   },
   \\"children\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"I am so green!\\"
   }
  },
  \\"createdBy\\": \\"Bojagi Collector\\",
  \\"name\\": \\"Green\\"
 },
 {
  \\"propSet\\": {
   \\"color\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"red\\"
   },
   \\"children\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"I am so green! I mean red...\\"
   }
  },
  \\"createdBy\\": \\"Bojagi Collector\\",
  \\"name\\": \\"Red\\"
 },
 {
  \\"propSet\\": {
   \\"color\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"blue\\"
   },
   \\"children\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"Blue.\\"
   }
  },
  \\"createdBy\\": \\"Bojagi Collector\\",
  \\"name\\": \\"Blue\\"
 },
 {
  \\"propSet\\": {
   \\"color\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"brown\\"
   },
   \\"children\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"Gasp\\"
   }
  },
  \\"createdBy\\": \\"Bojagi Collector\\",
  \\"name\\": \\"Brown\\"
 }
]"
`;

exports[`run main collector 6`] = `"ac41292a1db29019c2fbdac8701e94e2d50d45a5a7c1378cfb9f442f133df59e"`;

exports[`run main collector 7`] = `
"[
 {
  \\"propSet\\": {
   \\"color\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"white\\"
   },
   \\"buttons\\": {
    \\"type\\": \\"object\\",
    \\"value\\": [
     {
      \\"text\\": \\"stop\\",
      \\"color\\": \\"red\\"
     },
     {
      \\"text\\": \\"ready\\",
      \\"color\\": \\"yellow\\"
     },
     {
      \\"text\\": \\"go\\",
      \\"color\\": \\"green\\"
     }
    ]
   }
  },
  \\"createdBy\\": \\"Bojagi Collector\\",
  \\"name\\": \\"Traffic Light\\"
 }
]"
`;

exports[`run main collector 8`] = `"0895e4600fffb6dcbea59eda167c2ca6866e4bbc4dd8ade699cee61875d62c2b"`;

exports[`run main collector 9`] = `
"[
 {
  \\"propSet\\": {
   \\"color\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"yellow\\"
   },
   \\"children\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"click!\\"
   }
  },
  \\"createdBy\\": \\"Bojagi Collector\\",
  \\"name\\": \\"Button Yellow\\"
 },
 {
  \\"propSet\\": {
   \\"color\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"green\\"
   },
   \\"children\\": {
    \\"type\\": \\"string\\",
    \\"value\\": \\"click!\\"
   }
  },
  \\"createdBy\\": \\"Bojagi Collector\\",
  \\"name\\": \\"Button Green\\"
 }
]"
`;
